internal/pkg/server/agent.go

```diff
@@ -115,6 +115,42 @@ func (a *Agent) Run(ctx context.Context) error {
 		}
 		return p
 	})
+	a.agent.RegisterDiagnosticHook("fleet-server api tls diag", "fleet-server's API TLS config", "fleet-server-api-tls.txt", "text/plain", func() []byte {
+		if a.srv == nil {
+			log.Warn().Msg("Diagnostics hook failure fleet-server is nil.")
```

Comment:
If the content of the file is plain, why not just add this message into the file output.